### PR TITLE
SCP-2430 - Save contract nicknames locally in the browser.

### PR DESCRIPTION
**Pull request by merivale**
The main purpose of this PR is to make contract nicknames editable, and save these in localStorage. But while I was at it I tidied up the way localStorage things are handled, making the `ManageMarloweStorage` capability the only thing that interacts with localStorage directly. And removing the unwanted feature of prompting users to start using the wallet they were using last time (no longer saving this to localStorage).

I also made an unrelated improvement along the way - deleting some duplicate metadata lenses, and moving some others into web-common-marlowe.

### CODE REVIEW

File: marlowe-dashboard-client/src/Capability/MarloweStorage.purs [hunk 1]
```diff
@@ -49,26 +62,32 @@ class
   insertWalletRoleContracts :: String -> MarloweParams -> MarloweData -> m Unit
 
 instance manageMarloweStorageAppM :: ManageMarloweStorage AppM where
+  clearAllLocalStorage = do
```
Comment (hrajchert) [inline marlowe-dashboard-client/src/Capability/MarloweStorage.purs:65]: You can add a single liftEffect here

**Review by hrajchert — APPROVED** [review]
LGTM, but how are we gonna share the nickname with other participants? Or is it always local?

File: marlowe-dashboard-client/src/Capability/MarloweStorage.purs (hunk 1 above)
Comment (merivale) [reply marlowe-dashboard-client/src/Capability/MarloweStorage.purs:65]: ah yes, good point

**Comment by merivale** [comment]
Yeah, it's just going to be local for now. Simon and Russ really wanted them to be editable, and this is better than nothing. If we want to share the names, that's ultimately probably a job for the metadata server.

**Pull request merged**